common/kernel/indexed_store.h

```diff
@@ -209,6 +209,7 @@ template <typename T> class indexed_store
         return it;
     }
     iterator end() { return iterator{this, int32_t(slots.size())}; }
+    iterator rbegin() { return iterator{this, int32_t(slots.size()-1)}; }
```

Comment:
Hmm, this won't actually work if the last slot isn't occupied (because an item was removed). I think the original code was just an error, and derefencing normal `begin` instead of `rbegin` or `end` should work as it only cares about the one-user case anyway.

Reply:
Indeed `begin` is better.
But if the last slot may be unoccupied, the first slot may be unoccupied in a similar way, so we should use some other getter that returns the first occupied slot (not yet looked at the code for that).

Reply:
Yup, that's exactly how `begin` behaves - it will always return the first occupied slot, if there is one (which is guaranteed by the previous check in the case we're dealing with here)